lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -120,25 +126,32 @@ impl PointerTargetRegistry {
 
 #[derive(Clone, CloneRef, Debug)]
 pub struct Mouse {
-    pub mouse_manager: MouseManager,
-    pub last_position: Rc<Cell<Vector2<i32>>>,
-    pub position:      Uniform<Vector2<i32>>,
-    pub click_count:   Uniform<i32>,
-    pub hover_rgba:    Uniform<Vector4<u32>>,
-    pub target:        Rc<Cell<PointerTargetId>>,
-    pub handles:       Rc<[callback::Handle; 6]>,
-    pub frp:           enso_frp::io::Mouse,
-    pub scene_frp:     Frp,
+    pub mouse_manager:   MouseManager,
+    pub last_position:   Rc<Cell<Vector2<i32>>>,
+    pub position:        Uniform<Vector2<i32>>,
+    pub click_count:     Uniform<i32>,
+    pub hover_rgba:      Uniform<Vector4<u32>>,
+    pub target:          Rc<Cell<PointerTargetId>>,
+    pub handles:         Rc<[callback::Handle; 6]>,
+    pub scene_frp:       Frp,
+    pub last_move_event: Rc<RefCell<Option<mouse::Move>>>,
+    /// # Deprecated
+    /// This API is deprecated. Instead, use the display object's event API. For example, to get an
+    /// FRP endpoint for mouse event, you can use the [`crate::display::Object::on_event`]
+    /// function.
+    pub frp_deprecated:  enso_frp::io::Mouse_DEPRECATED,
```

Comment:
This structure became a bit convoluted. The responsibilities are not clear. It creates (and thus maintains) `target`, but never set the value. `hover_rgba` does not keep RGBA, as one could expect, only the target ID encoded as vector4. Again, it's created here, but never set nor read in this structure. The lack of documentation does not help.

There should be clear information, that `target` may be from a few frames ago (as it's updated from the pixel read pass).

Reply:
1. It uses `target` in the event listeners and manages the event listeners, so it kind of makes sense to store `target` here, but maybe we can find a better place.
2. `hover_rbga` - this is an old code, but this is a good catch. I renamed it to `pointer_target_encoded`.
3. Added docs to the above struct.